src/bundles/unittest/functions.ts

```diff
@@ -0,0 +1,82 @@
+import { TestContext, TestSuite, Test } from './types';
+
+const handleErr = (err: any) => {
+  if (err.error && err.error.message) {
+    return (err.error as Error).message;
+  }
+  if (err.message) {
+    return (err as Error).message;
+  }
+  throw err;
+};
+
+export const context: TestContext = {
+  describe: (msg: string, suite: TestSuite) => {
+    const starttime = performance.now();
+    context.suiteResults = {
+      name: msg,
+      results: [],
+      total: 0,
+      passed: 0,
+    };
+
+    suite();
+
+    context.allResults.results.push(context.suiteResults);
+
+    const endtime = performance.now();
+    context.runtime += endtime - starttime;
+    return context.allResults;
+  },
+
+  it: (msg: string, test: Test) => {
+    const name = `${msg}`;
+    let error = '';
+    context.suiteResults.total += 1;
+
+    try {
+      test();
+    } catch (err: any) {
+      error = handleErr(err);
+    }
```

Comment:
Is it possible to avoid mutation of the `error` variable here? Not really a huge problem in this function since scope of `error` is limited, but if function becomes bigger, it will start to become harder to keep track of all the mutations.

Reply:
There is also some mutations to other properties in this same `context` object which tbh I am uncomfortable with but I won't force u to change. But I would suggest minimising these mutations to reduce the number of bugs we may encounter in the future.

Reply:
`error` default to empty and is populated when an error is encoutered. I don't think this is a problem.